packages/hardhat/contracts/QuadraticDiplomacyContract.sol

```diff
@@ -40,8 +40,8 @@ contract QuadraticDiplomacyContract is Distributor, AccessControl {
 
         // remove all voter roles
         // an alternative (and less gas heavy?) way to do this would be to change the VOTER_ROLE to keccak256(abi.encodePacked(block.number));
-        for (uint256 i = 0; i < getRoleMemberCount(VOTER_ROLE); i++) {
-            revokeRole(VOTER_ROLE, getRoleMember(VOTER_ROLE, i));
+        while (getRoleMemberCount(VOTER_ROLE) > 0) {
```

Comment:
@sogasg Oh, I see! So the issue was that `getRoleMemberCount` was returning a different value on each call (since we were revoking members).

With this change we are still calling it **every time**. Out of curiosity, would this be better or worse (Thinking in terms of gas)?:

```solidity
uint256 memberCount = getRoleMemberCount(VOTER_ROLE);
for (uint256 i = 0; i < memberCount; i++) {
  revokeRole(VOTER_ROLE, getRoleMember(VOTER_ROLE, i));
}
```

Reply:
You could do it that way if we used getRoleMember(VOTER_ROLE, 0). But in that case we would not be using the i variable. So I thought it was cleaner to use while then.

Also, I am not sure, but maybe we save some gas on not instantiating menberCount or i 🤔

Reply:
On second thought, let us do it as you propose, but just ignoring the i and setting the index to get to 0. That will probably be the cheapest as long as there is more than two voters at least 👏